lib/plugins/aws/package/compile/events/eventBridge/index.js

```diff
@@ -12,13 +12,29 @@ class AwsCompileEventBridgeEvents {
     this.provider = this.serverless.getProvider('aws');
 
     this.hooks = {
+      'initialize': () => {
+        const hasFunctionsWithEventBridgeTrigger = Object.values(
+          this.serverless.service.functions
+        ).some(({ events }) => events.some(({ eventBridge }) => eventBridge));
+        const shouldUseCloudFormation =
+          this.serverless.service.provider.eventBridge &&
+          this.serverless.service.provider.eventBridge.useCloudFormation;
```

Comment:
We can be more optimal here, and check on this property in first place. And eventually return early before doing any deep functions check